docs/ProblemSolving.md

```diff
@@ -89,3 +89,88 @@ if tt_tensors_device.layout == ttnn.TILE_LAYOUT:
 Find where the lowering for the given operation is located and ensure that the arguments match the expected signature. 
 In this case, make sure that memory_config and dtype are passed as keyword arguments.
 
+## Backend Compiler Failure
+
+If you see an error similar to the following:
+
+```
+E                   torch._dynamo.exc.BackendCompilerFailed: backend='ttnn_backend' raised:
+E                   Exception: An error occurred when running the 'ToTtPass' pass after the following passes: []
+
+../tt-metal/python_env/lib/python3.8/site-packages/torch/fx/passes/infra/pass_manager.py:296: BackendCompilerFailed
+```
+
+### Debugging steps
+Run the test again but include `--trace`.
+
+
+Take note of the source file and line number where `BackendCompilerFailed` happened in the error above. Set a breakpoint there:
+
+```
+(Pdb) b ../tt-metal/python_env/lib/python3.8/site-packages/torch/fx/passes/infra/pass_manager.py:296
+```
+***
+```
+Breakpoint 1 at /home/ubuntu/repo/tt-metal/build/python_env/lib/python3.8/site-packages/torch/fx/passes/infra/pass_manager.py:296
+```
+
+Run the test and wait for the breakpoint to be hit:
+```
+(Pdb) r
+```
+***
+```
+> /home/ubuntu/repo/tt-metal/build/python_env/lib/python3.8/site-packages/torch/fx/passes/infra/pass_manager.py(296)__call__()
+-> raise Exception(msg) from e
+```
+
+Run interactive mode:
+```
+(Pdb) interact
+```
+***
+```
+*interactive*
+```
+Retrieve the source of the error:
+```
+>>> import pdb
+>>> pdb.post_mortem(e.__traceback__)
+```
+***
+```
+> /home/ubuntu/repo/tt-metal/build/python_env/lib/python3.8/site-packages/torch/fx/node.py(98)_get_qualified_name()
+-> raise RuntimeError("Unable to represent lambda") from e
+```
+
+## Incompatible Function Arguments
```

Comment:
> Incompatible Function Arguments

In this way we now have 2 sections with the same name, right?
Lets distinguish between wrong arguments passed (like position vs kw vs missing vs type mismatch).

This particular one is on a type mismatch, but more specifically for Tensor type mismatch, which requires a special handling.

Reply:
Done. Renamed to `Mismatch Between Torch and TTNN Tensor Types`